src/Models/ModelInstantiator.php

```diff
@@ -48,7 +49,8 @@ public function arrayToDataObject(
                 !array_key_exists($property, $cardinalities)
             )
             {
-                $instance->{$property} = $arr[$property];
+                $property_reflection = new ReflectionProperty($class, $property);
+                $instance->{$property} = $property_reflection->getType()->getName() === 'DateTime' ? new DateTime($arr[$property]) : $arr[$property];
```

Comment:
What about `null`?  If a `DateTime` is nullable, `null` should be copied to the property instead of `new DateTime(null)`, right?